### DISCUSSION TITLE
Login using SSO from frontend not working

### DISCUSSION THREAD

**Discussion by u12206050** [directus/directus#15357] [Archive]
### Describe the Bug

When implementing login flow using a SSO provider as described here: https://docs.directus.io/reference/authentication.html#login-using-sso-providers

The initial authentication to Auth0 seems to go well, until the callback to `https://directus-instance.com/auth/login/fake_provider/authorize`

### To Reproduce

From a SPA app try becoming authenticated using the Directus backend to authenticate with an SSO provider.

Using `@auth0/auth0-spa-js` this is my initial client:

```js
const client = await createAuth0Client({
    domain: "https://directus-instance.com/auth/login/fake_provider",
})

client.loginWithRedirect({ appState: { targetUrl: window.location.origin } })
```

### Errors Shown

`{"errors":[{"message":"Route /auth/login/fake_provider/authorize doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}`

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

16.4.0

### What database are you using?

mysql 8

### What browser are you using?

chromium

### How are you deploying Directus?

Locally & GCP

**Comment by rijkvanzanten** [comment]
I think there's two things going wrong here. The first being that it's not `/authorize`, but `/callback`, and secondly, Directus is an OAuth2 _client_, not _provider_, so you wouldn't use an OAuth2 client (like `@auth0/auth0-spa-js`) to login to Directus on `https://directus-instance.com/auth/login/fake_provider`, but instead, you'd open `https://directus-instance.com/auth/login/fake_provider` directly in the browser (either with an `<a>`, or `window.location.href = x`.

**Reply by u12206050** [reply]
Yeh you are right, I found these docs that were better for what I was trying to do: https://docs.directus.io/self-hosted/sso.html#seamless-sso
